discord/message.go

```diff
@@ -102,6 +102,7 @@ type Message struct {
 	Type                 MessageType           `json:"type"`
 	Flags                MessageFlags          `json:"flags"`
 	MessageReference     *MessageReference     `json:"message_reference,omitempty"`
+	MessageSnapshot      *Message              `json:"message_snapshot,omitempty"`
```

Comment:
Not sure if you want to map onto `Message` directly here since the fields returned here are a subset. May or may not work out for this project depending on strictness (e.g. `author` is not returned, but is a required field on Message in general)

Reply:
> Not sure if you want to map onto `Message` directly here since the fields returned here are a subset. May or may not work out for this project depending on strictness (e.g. `author` is not returned, but is a required field on Message in general)

that's why I mentioned clarification on the fields present in the PR description- it'd be helpful to know to perhaps make a custom type for fields that are guaranteed to be in the snapshots, if there are only a handful